FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpMsg.kt

```diff
@@ -65,4 +81,198 @@ object PgpMsg {
   )
 
   private val maxTagNumber = 20
+
+  enum class DecryptionErrorType {
+    KEY_MISMATCH,
+
+    // skip, effective for symmetric encryption
+    // USE_PASSWORD,
+
+    WRONG_PASSWORD,
+    NO_MDC,
+    BAD_MDC,
+
+    // a passphrase was not provided for the key which requires it,
+    // not sure how to detect this in the PGPainless.
+    NEED_PASSPHRASE,
+
+    FORMAT,
+    OTHER
+  }
+
+  data class DecryptionError(
+      val type: DecryptionErrorType,
+      val message: String,
+      val cause: Throwable? = null
+  )
+
+  data class DecryptionResult(
```

Comment:
Please look at this data class. It seems some of the fields can be null by default(for example `filename`). In that case we can a little shorter variant. For example

```
return DecryptionResult(
            isEncrypted = false,
            error = DecryptionError(type, message, cause)
        )
```

Reply:
yes, can be made better, will update it

Reply:
:+1: